### ISSUE TITLE
3rd Party Libraries Not Required for Recaptcha Example, But They Complicate Implementation of Recaptcha

### ISSUE THREAD

**Issue by danallendotcom** [google/recaptcha#54]
The example works perfectly.  It's even better after removing doctrine and reflection-docblock from the vendor directory.

Call me a retard, but the fact that the example at examples/example-captcha.php runs without doctrine and reflection-docback makes me think those libraries are not required.

Why are those libraries included in this sample?  

I really don't want to add libraries that are  not needed.  I am bringing this up as an issue because the sample code should include only essential code, to avoid propagating libraries for no reason.

**Comment by danallendotcom** [comment]
Actually, the demo works perfectly with the following line commented out:
file:  examples/example-captcha.php
line number: 36
CHANGE FROM 
require_once __DIR__ . '/../vendor/autoload.php';

CHANGE TO 
//require_once __DIR__ . '/../vendor/autoload.php';
require_once __DIR__ . '/../src/ReCaptcha/ReCaptcha.php';
require_once __DIR__ . '/../src/ReCaptcha/RequestMethod.php';
require_once __DIR__ . '/../src/ReCaptcha/RequestMethod/Post.php';
require_once __DIR__ . '/../src/ReCaptcha/RequestParameters.php';
require_once __DIR__ . '/../src/ReCaptcha/Response.php';


In other words, the whole vendor/autoload/composer deal should be dropped from the demo, since it add overheard but is not required for correct functioning of the demo.

**Comment by danallendotcom** [comment]
If there is a good reason for including 3rd party libraries in this project, it would be my pleasure to mark this issue invalid and close it.  Same applies to any reason that would make this an invalid issue.  If there isn't a good reason for the 3rd party libraries, maybe that isn't for anybody but me.    So, is this a valid issue or not?

**Comment by haidongs** [comment]
I believe we could just remove the 3rd party libraries.

-Haidong

On Thu, Jun 25, 2015 at 2:08 PM, danallendotcom <notifications@github.com>
wrote:

> If there is a good reason for including 3rd party libraries in this
> project, it would be my pleasure to mark this issue invalid and close it.
> Same applies to any reason that would make this an invalid issue. If there
> isn't a good reason for the 3rd party libraries, maybe that isn't for
> anybody but me. So, is this a valid issue or not?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/recaptcha/issues/54#issuecomment-115398941>.
>



-- 

Haidong Shao |  | haidongs@google.com | 4086005401

**Comment by noxidsoft** [comment]
I think I'll just remove reCaptcha from my implementation all together, too hard to upgrade/implement at the moment.

**Comment by danallendotcom** [comment]
noxinsoft, This branch is significant simplification, that includes 100% of reCaptha, but leaves out components unrelated to reCaptcha.  Is this something you can work with?  I think it is pretty simple in this form.  
https://github.com/danallendotcom/recaptcha/tree/streamlined

If there anything you think of that could make it simpler or better, I hope you will let me know so I can address it.

**Comment by rowan-m** [comment]
@danallendotcom there are no required 3rd party dependencies.

e.g.
```
$ mkdir myproject
$ cd myproject
$ composer init -n --name="myvendor/myproject" --require="google/recaptcha ~1.1"
$ composer install
$ find . | wc -l
```
This will show you a total of 41 files. Composer itself accounts for 6 of those.

If you wish to use the project without Composer, then you can use the  [Manual Install](https://github.com/google/recaptcha#manual-installation) instructions. That is, you can either clone or download the repo and use the autoloader script to pull in the classes.

Alternatively, you can also write your own autoloader or just manually include classes directly in your code.

It sounds to me like you've cloned the repo and then run `composer install` from there. By default, this pulls in the development dependencies. In this case, that's PHPUnit and its dependencies required to run the unit tests. You can avoid this by running `composer install --no-dev` instead.

**Comment by rowan-m** [comment]
@danallendotcom @noxidsoft I've updated the README to clarify the functionality. Please let me know if that clears things up for you.

**Comment by danallendotcom** [comment]
"Hey man, i wrote a memo explaining how to clean the blood running down your body where the knife is in your side.  I hope that solves the problem with the kniife."

This issue is not resolved.  I do not understand why it has been closed.  The information recently posted does not change the fact that most of this repository is extraneous and confusing.  

This project is configured for the convenience of Google developers when it is also the delivery channel to millions of web developers globally.  Google has a huge splash at https://www.google.com/recaptcha/, saying recaptcha is easy for humans.  That splash includes a Developers Guilde that says "Here, we explain how to add reCAPTCHA to your page,"    and a link is provided to this project as an example of how to do it. The instrucitions outside this example are incoherent, leaving this working model as the only ray of light toward implementation.   https://developers.google.com/recaptcha/intro

This project is not appproproiate for that level of public consumption.  The guide says "This documentation is designed for people familiar with HTML forms and server-side processing."  I am familiar with those things.  In fact, I am very familair with those things.  If you really want to push the "you have to know what you are doing" card, I'll bet I am as familiar with html and server-side processing as anyone on your team.  While this project might seem intuitive to you, the person who created it, it is not intuitive to anyone not familiar with Composer. 

Have you put yourself in the shoes of the developers aroumd the world whose bosses are looking at a a checkbox and "easy on humans" who want their developers to implement this, who are then wondering why it takes more than minute to figure this thing out, because they have not been exposed to Compsoer?  Composer is not a requirement for recaptha.  Includimg it in this release creates a big problem for people who have not used it.  

Instead of explaining it, it should be removed.

Let me restate the issue:  EXTRANEOUS CONFUSING CONFIGURATION

Explaining it does not fix it.

I suspect you don't want to fix this problem because the extraneous material creates a perception of there being more to this than there is, somehow generating a sense of importance.

**Comment by WouterDS** [comment]
@danallendotcom Hey man, ease up a bit. I'm a 22 year old developer with a few years of experience and I was able to set it up client side & server side validation within half an hour. I found everything I needed on https://developers.google.com/recaptcha/docs/start.

And people that are doing web development and don't know what composer is (the easiest dependency manager ever) are just behind and need to update their skills. Recaptcha can be used without it as mentioned above, but demanding to remove composer from the repository because you don't like it is not going to happen.

**Comment by danallendotcom** [comment]
wouter, everyone is behind in many basic aspects of our field, no one is
current on everything importantant.  Too many things develop too fast for
any human to keep up in it all. That makes it important to exclude
extraneous elements.

Are you thinking you are current on everything that matters in our field?
Well, since you mentioned how easy it was for you, I might as well admit I
am a dumbass.  I graduated from MIT before a lot of the technology we use
today was invented.  If I were smart, like you , I would have postponed my
education until the modern things were invented.

Whatever you or I have had time to learn has come at the expense of not
having time to learn something else.  You should back off telling me to
take it easy. Since I  am behind in composer, as a result of not having
needed it before , it took me a day to do reading up and experimenting
necessary for me to figure out how to implement recaptha.  When I
discovered after all that no 3rd party libraries are required, I started
wondering what dickheads were putting the software out in such a confusing
manner.  I was glad enough to get an education in Composer, but
my preference is to put my time into areas that move my project forward.
In this case, Composer's presence wasted my time.  Techies are notorious
for making things too complicated, I do that too. Instead of defending it,
we should fix it.

I am glad we were able to straighten out this problem.  Please let me know
if you have any more questions.  I  always glad to help out beginners.

On Tuesday, July 7, 2015, Wouter De Schuyter <notifications@github.com>
wrote:

> @danallendotcom <https://github.com/danallendotcom> Hey man, ease up a
> bit. I'm a 22 year old developer with a few years of experience and I was
> able to set it up client side & server side validation within half an hour.
> I found everything I needed on
> https://developers.google.com/recaptcha/docs/start.
>
> And people that are doing web development and don't know what composer is
> (the easiest dependency manager ever) are just behind and need to update
> their skills. Recaptcha can be used without it as mentioned above, but
> demanding to remove composer from the repository because you don't like it
> is not going to happen.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/recaptcha/issues/54#issuecomment-119307356>.
>


-- 
Dan Allen
Houston-DATA.COM
215 Marshall Street
Houston, Texas 77006
713-806-8783
dan@houston-data.com

**Comment by sergey-serov** [comment]
*"Too many things develop too fast for any human to keep up in it all. That makes it important to exclude extraneous elements."*

**Good words, imho.**
